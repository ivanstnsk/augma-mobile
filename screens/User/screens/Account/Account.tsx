import * as React from 'react';
import { StyleSheet, View } from 'react-native';

import { Button } from 'components/Button';
import { Text } from 'components/Text';
import { Avatar } from 'components/Avatar';
import { useUserInfo } from 'store/user';
import { useLogin } from 'hooks';

export const Account: React.FC = () => {
  const Login = useLogin();
  const UserInfo = useUserInfo();

  const handleLogout = () => {
    Login.logout();
  }

  return (
    <View style={styles.wrapper}>
      <View style={styles.infoContainer}>
        <Avatar />
        <Text style={styles.name}>{UserInfo.name}</Text>
      </View>
      <Button
        onPress={handleLogout}
        label="Logout"
        variant="normal"
        color="primary"
      />
    </View>
  );
}

const styles = StyleSheet.create({
  wrapper: {
    flex: 1,
    flexDirection: 'column',
    alignItems: 'stretch',
    justifyContent: 'flex-start',
    backgroundColor: '#1F1F1F',
    paddingHorizontal: 20,
    paddingTop: 16,
    paddingBottom: 19,
  },
  infoContainer: {
    flex: 1,
    flexDirection: 'column',
    alignItems: 'center',
  },
  name: {
    marginTop: 24,
    fontSize: 18,
  }
});
