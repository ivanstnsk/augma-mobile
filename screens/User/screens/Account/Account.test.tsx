import React from 'react';
import renderer from 'react-test-renderer';

import { Account } from './Account';

jest.mock('hooks', () => ({
  useLogin: jest.fn().mockReturnValue({
    logout: jest.fn(),
  })
}));

jest.mock('store/user', () => ({
  useUserInfo: jest.fn().mockReturnValue({
    name: 'Test',
  })
}));

jest.mock('components/Text', () => ({
  Text: () => "Text"
}));

jest.mock('components/Avatar', () => ({
  Avatar: () => "Avatar"
}));

jest.mock('components/Button', () => ({
  Button: () => "Button"
}));

describe('Account', () => {
  it('should render Account', () => {
    const tree = renderer.create(<Account />).toJSON();
   
    expect(tree).toMatchSnapshot();
  });
});