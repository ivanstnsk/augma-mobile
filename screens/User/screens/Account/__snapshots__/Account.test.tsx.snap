// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account should render Account 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#1F1F1F",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "paddingBottom": 19,
      "paddingHorizontal": 20,
      "paddingTop": 16,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    Avatar
    Text
  </View>
  Button
</View>
`;
